cpp/src/arrow/dataset/dataset.h

```diff
@@ -90,6 +90,8 @@ class ARROW_DS_EXPORT Fragment : public std::enable_shared_from_this<Fragment> {
 
   virtual ~Fragment() = default;
 
+  bool handles_compute = true;
```

Comment:
Can we document what this flag means?

Reply:
This is still unclear to me without having to go look at the implementation - does `true` mean this fragment applies filters itself, or no?